benchmarking/notebooks/analyze-results.ipynb

```diff
@@ -71,82 +68,123 @@
   },
   {
    "cell_type": "code",
-   "execution_count": null,
-   "id": "2707495e",
+   "execution_count": 4,
+   "id": "a7e143d3",
    "metadata": {},
    "outputs": [],
    "source": [
-    "def plot_client_vs_server_barplots(df_user: pd.DataFrame, x_col: str, y_cols: List[str], title: str, ylabel: str) -> None:\n",
+    "def plot_client_vs_server_barplots(df_user: pd.DataFrame, x_col: str, y_cols: List[str], legend_labels: List[str], title: str, ylabel: str, xlabel: str) -> None:\n",
     "    \"\"\"\n",
     "    Plots bar plots for client vs server metrics from a DataFrame.\n",
     "\n",
     "    Args:\n",
     "        df_user (pd.DataFrame): The DataFrame containing the data to plot.\n",
     "        x_col (str): The column name to be used as the x-axis.\n",
     "        y_cols (List[str]): A list of column names to be used as the y-axis.\n",
+    "        legend_labels (List[str]): Human-readable labels for each grouping in y_cols.\n",
     "        title (str): The title of the plot.\n",
     "        ylabel (str): The label for the y-axis.\n",
+    "        xlabel (str): The label for the x-axis.\n",
     "\n",
     "    Returns:\n",
-    "        None\n",
+    "        fig (go.Figure): The plotly figure container\n",
     "    \"\"\"    \n",
-    "    # Melt the DataFrame to have a long-form DataFrame suitable for Seaborn\n",
-    "    df_melted = df_user.melt(id_vars=[x_col], value_vars=y_cols, \n",
-    "                             var_name='Metric', value_name='Value')\n",
+    "    value_vars = y_cols\n",
+    "    title_text = title\n",
+    "    yaxis_title = ylabel\n",
+    "    xaxis_title = xlabel\n",
     "\n",
-    "    # Create the plot\n",
-    "    _, ax = plt.subplots(figsize=(10, 6))\n",
-    "    sns.barplot(data=df_melted, x=x_col, y='Value', hue='Metric', ax=ax, estimator=np.median)\n",
+    "    df_melted = df_user.melt(\n",
+    "        id_vars=[x_col], \n",
+    "        value_vars=value_vars, \n",
+    "        var_name='Metric', \n",
+    "        value_name='Value',\n",
+    "    )\n",
+    "    xgroups = [str(x) for x in sorted(pd.unique(df_melted[x_col]))]\n",
+    "    df_melted[x_col] = [str(x) for x in df_melted[x_col]]\n",
     "\n",
-    "    # Customize the plot\n",
-    "    plt.title(title)\n",
-    "    plt.xlabel('Batch Size Used')\n",
-    "    plt.ylabel(ylabel)\n",
+    "    fig = go.Figure()\n",
+    "    for i in xgroups:\n",
+    "        showlegend = i == xgroups[0]\n",
+    "        fig.add_trace(go.Violin(\n",
```

Comment:
I've uploaded some "better" plots that take into account our conversation on Friday.

Worth noting that the "Distribution of throughput by batch size" and the "Total throughput by batch size" graphs now tread VERY similar territory: the former deals in percentile-based stats (5th/50th/95th percentile), while the latter simply reports the mean stat (first moment).

As this is a subtlety that will likely be lost on the not-too-stats-savvy users whom we are now targeting, I'd recommend dropping the latter chart entirely as (nearly) redundant. Your call.

Reply:
hi @snova-michaelw they are similar when bs=1 but the result is different for other bs like 4,8,16. I think we should still keep it since it shows the potential of tokens/s processed per each batch in total (ie formula: throughput*batch_size)